File: tests/phpunit/CollationTest.php
<?php

namespace Ecma\Test\Intl;

use Ecma\Intl\Collation;

class CollationTest extends TestCase
{
    /**
     * @dataProvider collationValuesProvider
     */
    public function testCollationValues(Collation $collation, string $expectedValue): void
    {
        $this->assertSame($expectedValue, $collation->value);
    }

    /**
     * @return array<array{collation: Collation, expectedValue: string}>
     */
    public function collationValuesProvider(): array
    {
        return [
            [
                'collation' => Collation::Big5han,
                'expectedValue' => 'big5han',
            ],
            [
                'collation' => Collation::Compat,
                'expectedValue' => 'compat',
            ],
            [
                'collation' => Collation::Dict,
                'expectedValue' => 'dict',
            ],
            [
                'collation' => Collation::Direct,
                'expectedValue' => 'direct',
            ],
            [
                'collation' => Collation::Ducet,
                'expectedValue' => 'ducet',
            ],
            [
                'collation' => Collation::Emoji,
                'expectedValue' => 'emoji',
            ],
            [
                'collation' => Collation::Eor,
                'expectedValue' => 'eor',
            ],
            [
                'collation' => Collation::Gb2312,
                'expectedValue' => 'gb2312',
            ],
            [
                'collation' => Collation::Phonebk,
                'expectedValue' => 'phonebk',
            ],
            [
                'collation' => Collation::Phonetic,
                'expectedValue' => 'phonetic',
            ],
            [
                'collation' => Collation::Pinyin,
                'expectedValue' => 'pinyin',
            ],
            [
                'collation' => Collation::Reformed,
                'expectedValue' => 'reformed',
            ],
            [
                'collation' => Collation::Search,
                'expectedValue' => 'search',
            ],
            [
                'collation' => Collation::Searchjl,
                'expectedValue' => 'searchjl',
            ],
            [
                'collation' => Collation::Standard,
                'expectedValue' => 'standard',
            ],
            [
                'collation' => Collation::Stroke,
                'expectedValue' => 'stroke',
            ],
            [
                'collation' => Collation::Trad,
                'expectedValue' => 'trad',
            ],
            [
                'collation' => Collation::Unihan,
                'expectedValue' => 'unihan',
            ],
            [
                'collation' => Collation::Zhuyin,
                'expectedValue' => 'zhuyin',
            ],
        ];
    }
}

File: tests/criterion/php/support/collation_test.c
#include "src/php/support/collation.h"
#include "tests/criterion/test.h"

#define TEST_SUITE phpSupportCollation

ParameterizedTestParameters(TEST_SUITE, getsCollationCaseNameForValue) {
  START_BASIC_TEST_PARAMS(21)

  BASIC_TEST("big5han", "Big5han")
  BASIC_TEST("compat", "Compat")
  BASIC_TEST("dict", "Dict")
  BASIC_TEST("direct", "Direct")
  BASIC_TEST("ducet", "Ducet")
  BASIC_TEST("emoji", "Emoji")
  BASIC_TEST("eor", "Eor")
  BASIC_TEST("gb2312", "Gb2312")
  BASIC_TEST("phonebk", "Phonebk")
  BASIC_TEST("phonetic", "Phonetic")
  BASIC_TEST("pinyin", "Pinyin")
  BASIC_TEST("reformed", "Reformed")
  BASIC_TEST("search", "Search")
  BASIC_TEST("searchjl", "Searchjl")
  BASIC_TEST("standard", "Standard")
  BASIC_TEST("stroke", "Stroke")
  BASIC_TEST("trad", "Trad")
  BASIC_TEST("unihan", "Unihan")
  BASIC_TEST("zhuyin", "Zhuyin")
  BASIC_TEST("foo", NULL)
  BASIC_TEST(NULL, NULL)

  END_BASIC_TEST_PARAMS;
}

BASIC_PARAMETERIZED_TEST(getsCollationCaseNameForValue, getCollationCaseName)
